src/viadot/orchestration/prefect/utils.py

```diff
@@ -28,6 +28,394 @@
+        replacements = {
+            "today": today.strftime(self.dynamic_date_format),
+            "yesterday": yesterday.strftime(self.dynamic_date_format),
+            "this month": today.strftime("%m"),
+            "last month": f"{last_month:02d}",
+            "this year": today.strftime("%Y"),
+            "last year": last_year.strftime("%Y"),
+            "now time": now_time.strftime("%H%M%S"),
+            "last day previous month": last_day_prev_month.strftime(
+                self.dynamic_date_format
+            ),
+        }
+        return replacements
+
+    def handle_singular_dates(self, match: List[tuple], key: str) -> pendulum.DateTime:
+        """
+        Directs execution of a specific function based on the provided value of `key`.
+        Returns a pendulum.DateTime date generated based on `match`.
+
+        Args:
+            match (List[tuple]): List of every pattern match that occurs in a given string.
+            key (str): Key phrase that determines the execution of a specific function.
+
+        Returns:
+            pendulum.DateTime: A dynamically created date.
+        """
+        if key == "last_day_of_month":
+            for month_name, year in match:
+                replacement = self.process_last_day_of_month(year, month_name)
+
+        elif key == "x_units_ago_full_date":
+            for x, units in match:
+                x = int(x)
+                replacement = self.get_date_x_ago_full_date(int(x), units)
+
+        elif key == "x_years_ago":
+            for x in match:
+                x = int(x)
+                replacement = self.process_x_years_ago(x)
+        return replacement
+
+    def generate_dates_based_on_unit(self, number: int, unit: str) -> List[str]:
+        """
+        Direct execution of a specific function based on the provided value of `unit`.
+        Returns a list of dynamically created dates generated based on `unit` in an ascending order.
+
+        Possible values of `unit` correspond to different date formatting styles:
+            - 'years': Return a date formatted with a pendulum token 'Y'.
+            - 'months': Return a date formatted with a pendulum token 'YMM'.
+            - 'days': Return a date with a pendulum token 'YMMDD'.
+        Args:
+            number (int): The number of units from the current year to include.
+            unit (str): The unit of time ('years', 'months', 'days').
+
+        Returns:
+            List[str]: A list of dates in ascending order.
+        """
+        if unit == "years":
+            return self.generate_years(
+                last_years=number, from_year=None, num_years=None
+            )
+        elif unit == "months":
+            return self.generate_months(last_months=number)
+        elif unit == "days":
+            return self.generate_dates(last_days=number)
+
+    def handle_data_ranges(self, match: List[tuple], key: str) -> List[str]:
```

Comment:
mark this function internal by adding `_`.
```suggestion
    def _handle_data_ranges(self, match: List[tuple], key: str) -> List[str]:
```